pkg/provisioner/ironic/update_opts.go

```diff
@@ -72,6 +73,27 @@ func deref(v interface{}) interface{} {
 	return ptrVal.Elem().Interface()
 }
 
+func sanitisedValue(data interface{}) interface{} {
+	dataType := reflect.TypeOf(data)
+	if dataType.Kind() != reflect.Map ||
+		dataType.Key().Kind() != reflect.String {
+		return data
+	}
+
+	value := reflect.ValueOf(data)
+	safeValue := reflect.MakeMap(dataType)
+
+	for _, k := range value.MapKeys() {
+		safeDatumValue := value.MapIndex(k)
+		if strings.Contains(k.String(), "password") {
```

Comment:
Just to be on the safe side, the comparison could be case insensitive

Reply:
This covers all of the values that any of the bmc drivers use.

Reply:
I was referring to the keys, which IINW are defined in the code upper func calls (like `SetPropertiesOpts`). So maybe someone could define a `Password` prop by mistake (not sure if Ironic is also strict on the case) which will not get sanitised

Reply:
Probably a `strings.ToLower(k.String())` wouldn't hurt, but if Ironic will catch it (ie during the integration tests) then it'd be ok

/lgtm